src/main/java/sk/uniba/fmph/dcs/game_board/ToolMakerHutFields.java

```diff
@@ -0,0 +1,185 @@
+    private PlayerOrder fieldsFigures;
+
+    private final int restriction; // for less than 4 players
+    private static final int MIN_PL = 2;
+    private static final int MAX_PL = 4;
+
+    public ToolMakerHutFields(int players){
+        this.restriction = players;
+        if (!(MIN_PL <= players && players <= MAX_PL)){
+            throw new IllegalArgumentException("Invalid number of players");
+        }
+        toolMakerFigures = null;
+        hutFigures = null;
+        fieldsFigures = null;
+    }
+
+    private boolean checkRestrictions()
+    {
+        if(this.restriction == MAX_PL)
+            return true;
+
+        int alreadyOccupiedLocations = 0;
+
+        if(toolMakerFigures != null)
+            alreadyOccupiedLocations++;
+        if(hutFigures != null)
+            alreadyOccupiedLocations++;
+        if(fieldsFigures != null)
+            alreadyOccupiedLocations++;
+
+        if(alreadyOccupiedLocations >= 2)
+            return false;
+        return true;
+    }
+
+    public boolean canPlaceOnToolMaker(Player player){
+        if(!checkRestrictions())
+            return false;
+        return toolMakerFigures == null && player.getPlayerBoard().hasFigures(1);
+    }
+
+    public boolean placeOnToolMaker(Player player)
+    {
+        if(canPlaceOnToolMaker(player)){
+            player.getPlayerBoard().takeFigures(1);
+            toolMakerFigures = player.getPlayerOrder();
+            return true;
+        }
+        return false;
+    }
+
+    public boolean canActionOnToolMaker(Player player){
+        return toolMakerFigures.equals(player.getPlayerOrder());
+    }
+
+    public boolean actionToolMaker(Player player){
+        if (canActionOnToolMaker(player)) {
+            Collection<Effect> newTool = new ArrayList<Effect>();
+            newTool.add(Effect.TOOL);
+            player.getPlayerBoard().giveEffect(newTool);
+            toolMakerFigures = null;
+            return true;
+        }
+        return false;
+    }
+
+    public boolean skipActionOnToolMaker(Player player){
+        if (canActionOnToolMaker(player)) {
+            player.getPlayerBoard().giveFigure();
+            toolMakerFigures = null;
+            return true;
+        }
+        return false;
+    }
+
+    boolean canPlaceOnHut(Player player){
+        if(!checkRestrictions())
+            return false;
+        return hutFigures == null  && player.getPlayerBoard().hasFigures(2);
+    }
+
+    public boolean placeOnHut(Player figures){
+        if(canPlaceOnHut(figures)){
+            figures.getPlayerBoard().takeFigures(2);
+            hutFigures = figures.getPlayerOrder();
+            return true;
+        }
+        return false;
+    }
+
+    public boolean canActionOnHut(Player player){
+        return hutFigures.equals(player.getPlayerOrder());
+    }
+
+    public boolean actionHut(Player player){
+        if (hutFigures.equals(player.getPlayerOrder())) {
+            player.getPlayerBoard().giveFigure();
```

Comment:
On this place we need giveFigure, but I agree that I need to delete it from skipActionOnX methods, because it is already implemented in PlayerBoard logic